src/sql/workbench/api/node/extHostNotebook.ts

```diff
@@ -51,7 +51,7 @@ export class ExtHostNotebook implements ExtHostNotebookShape {
 		let manager = this.findManagerForUri(uriString);
 		if (manager) {
 			manager.provider.handleNotebookClosed(uri);
-			// Note: deliberately not removing handle.
+			this._adapters.delete(manager.handle);
```

Comment:
I don't think this is the right approach. There are a few reasons for not removing the handle:
- Manager may receive server-level events independent of this. These will fail
- In the future, we should reduce total number of Jupyter sessions. In particular it should be 1 per folder / 1 per workspace if files are inside an ADS workspace. In this case closing a single notebook shouldn't shut down the server
- Correct disposal is to be handled either by the extension or our front end, not by this layer. 

I think we should take a careful look at this path and decide how to proceed. Doing this right now is a bit dangerous and so let's chat about what scenarios we need to fix and look at alternatives.

Reply:
Thanks Kevin. Could you explain more on "Manager may receive server-level events independent of this. These will fail"? So we could know what they are?